packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -776,7 +804,7 @@ export class ProjectService {
 
         // Force refresh adapter (refetch git repos, check for changed credentials, etc.)
         // Might want to cache parts of this in future if slow
-        const adapter = await this.restartAdapter(projectUuid);
+        const adapter = await this.buildAdapter(projectUuid);
         const packages = await adapter.getDbtPackages();
         try {
             const explores = await adapter.compileAllExplores();
```

Comment:
I think we need a destroy here as well.
```suggestion
            const explores = await adapter.compileAllExplores();
            await adapter.destroy();
```

Reply:
Also missing in the catch blocks for all the other places 🤔 I'm unsure about removing the adapter caching since we now need to destroy it in each method.